lightning-graph-sync/src/processing.rs

```diff
@@ -0,0 +1,565 @@
+use std::io;
+use std::io::Read;
+use std::time::{SystemTime, UNIX_EPOCH};
+
+use bitcoin::BlockHash;
+use bitcoin::secp256k1::PublicKey;
+
+use lightning::ln::msgs::{
+	DecodeError, ErrorAction, LightningError, OptionalField, UnsignedChannelUpdate,
+};
+use lightning::routing::network_graph;
+use lightning::util::ser::{BigSize, Readable};
+
+use crate::error::GraphSyncError;
+
+/// Backdating is used as a mechanism to avoid accidentally overriding higher-fidelity gossip
+/// updates. Current delta amount is set to a week flat. Note that gossip updates are purged at
+/// the two-week-mark.
+///
+/// In the future, we will probably start delaying the purging of unsigned updates to their
+/// four-week-marks, or perhaps simply trust a provided server-timestamp that marks
+/// data receipt, as opposed to originally signed value. TODO: TBD
+const TIMESTAMP_RETROGRESSION_DELTA: u32 = 7 * 24 * 3600;
+
+/// Update network graph from binary data
+///
+/// `network_graph`: network graph to be updated
+///
+/// `update_data`: `&[u8]` binary stream that comprises the update data
+pub fn update_network_graph(
+	network_graph: &network_graph::NetworkGraph,
+	update_data: &[u8],
+) -> Result<(), GraphSyncError> {
+	let mut read_cursor = io::Cursor::new(update_data);
+	read_network_graph(&network_graph, &mut read_cursor)
+}
+
+pub(crate) fn read_network_graph<R: Read>(
+	network_graph: &network_graph::NetworkGraph,
+	mut read_cursor: &mut R,
+) -> Result<(), GraphSyncError> {
+	let mut prefix = Vec::with_capacity(4);
+	prefix.resize(4, 0);
+	read_cursor.read_exact(&mut prefix)?;
+
+	let _deserialize_unsigned = match &prefix[..] {
+		&[76, 68, 75, 1] => true,
+		_ => {
+			return Err(GraphSyncError::ProcessingError(
+				"Prefix must equal 'LDK' and the version byte 1".to_string(),
+			));
+		}
+	};
+
+	// let chain_source: Option<Box<dyn lightning::chain::Access>> = None;
+	let current_timestamp = SystemTime::now()
+		.duration_since(UNIX_EPOCH)
+		.unwrap()
```

Comment:
nit: lol this is why we don't use rustfmt, lets not make our code so vertical please :)